src/internal/operations/backup_integration_test.go

```diff
@@ -191,67 +206,210 @@ func checkMetadataFilesExist(
+
+	items := make([]incrementalItem, 0, howMany)
 
-		paths = append(paths, p)
-		pathsByRef[dir.ShortRef()] = append(pathsByRef[dir.ShortRef()], fName)
+	for i := 0; i < howMany; i++ {
+		var (
+			now       = common.Now()
+			nowLegacy = common.FormatLegacyTime(time.Now())
+			id        = uuid.NewString()
+			subject   = "incr_test " + now[:16] + " - " + id[:8]
+			body      = "incr_test " + cat.String() + " generation for " + userID + " at " + now + " - " + id
+		)
+
+		items = append(items, incrementalItem{
+			name: id,
+			data: dbf(id, nowLegacy, subject, body),
+		})
 	}
 
-	cols, err := kw.RestoreMultipleItems(ctx, bup.SnapshotID, paths, nil)
-	assert.NoError(t, err)
+	collections := []incrementalCollection{{
+		pathElements: []string{destFldr},
+		category:     cat,
+		items:        items,
+	}}
 
-	for _, col := range cols {
-		itemNames := []string{}
+	dest := control.DefaultRestoreDestination(common.SimpleTimeTesting)
+	dest.ContainerName = destFldr
 
-		for item := range col.Items() {
-			assert.Implements(t, (*data.StreamSize)(nil), item)
+	dataColls, err := buildCollections(
+		t,
+		service,
+		tenantID, userID,
+		dest,
+		collections)
+	if err != nil {
+		return nil, err
+	}
 
-			s := item.(data.StreamSize)
-			assert.Greaterf(
-				t,
-				s.Size(),
-				int64(0),
-				"empty metadata file: %s/%s",
-				col.FullPath(),
-				item.UUID(),
-			)
+	return gc.RestoreDataCollections(ctx, sel, dest, dataColls)
+}
 
-			itemNames = append(itemNames, item.UUID())
-		}
+type incrementalItem struct {
+	name string
+	data []byte
+}
 
-		assert.ElementsMatchf(
+type incrementalCollection struct {
+	// Elements (in order) for the path representing this collection. Should
+	// only contain elements after the prefix that corso uses for the path. For
+	// example, a collection for the Inbox folder in exchange mail would just be
+	// "Inbox".
+	pathElements []string
+	category     path.CategoryType
+	items        []incrementalItem
+}
+
+func buildCollections(
+	t *testing.T,
+	service path.ServiceType,
+	tenant, user string,
+	dest control.RestoreDestination,
+	colls []incrementalCollection,
+) ([]data.Collection, error) {
+	t.Helper()
+
+	collections := make([]data.Collection, 0, len(colls))
+
+	for _, c := range colls {
+		pth, err := toDataLayerPath(
 			t,
-			pathsByRef[col.FullPath().ShortRef()],
-			itemNames,
-			"collection %s missing expected files",
-			col.FullPath(),
-		)
+			service,
+			tenant,
+			user,
+			c.category,
+			c.pathElements,
+			false)
+		if err != nil {
+			return nil, err
+		}
+
+		mc := mockconnector.NewMockExchangeCollection(pth, len(c.items))
+
+		for i := 0; i < len(c.items); i++ {
+			mc.Names[i] = c.items[i].name
+			mc.Data[i] = c.items[i].data
+		}
+
+		collections = append(collections, mc)
 	}
+
+	return collections, nil
+}
+
+func toDataLayerPath(
+	t *testing.T,
+	service path.ServiceType,
+	tenant, user string,
+	category path.CategoryType,
+	elements []string,
+	isItem bool,
+) (path.Path, error) {
```

Comment:
Not sure how much we are leveraging the `testing.T` item, but we could lose the error return if `require` was used. Additionally, there are several `toDataLayer`-type names within path, would it be beneficial to refactor this to `fetchTestPath` or something of the like?

Reply:
Per error vs assert/require: sure, that sounds good.

Per the `toDataLayer`, I've talked with ash about a QoL constructor in paths.  The idea has grounds, but isn't of much importance.  Might make a tech debt issue for it later.